.github/workflows/ruff.yml

```
@@ -0,0 +1,31 @@
+      - main
+  push:
+    branches:
+      - main
+
+jobs:
+    ruff:
+        name: Linter and format python code
+        runs-on: ubuntu-latest
+        steps:
+          - name: Check out the repo
+            uses: actions/checkout@v2
```

actionlint (1.7.9)

the runner of "actions/checkout@v2" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)